cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx

```diff
@@ -112,50 +119,55 @@ const Rule: React.FC<IRuleProps> = ({
-            className={classes.deleteButton}
-            onClick={() => deleteRule(ruleIdx)}
-          >
-            <CancelIcon fontSize="large" />
-          </IconButton>
-        </If>
-        {rule.map((stage, i) => (
-          <div key={`${i}-${stage.stageName}`} className={classes.stageRow}>
-            <div className={classes.tableCell}>{stage.stageName}</div>
-            <div className={classes.tableCell}>
-              <OutlinedSelect
-                options={inputSchema[stage.stageName]}
-                disabled={disabled}
-                value={stage.fieldName}
-                onChange={(val) => fieldChange(i, val)}
-              />
-            </div>
-            <If condition={i !== rule.length - 1}>
+    <div className={classes.root}>
+      <div
+        className={classes.ruleContainer}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+      >
+        <Paper className={classes.rule}>
+          <If condition={hovered && rulesCount !== 1 && !disabled}>
+            <IconButton
+              size="small"
+              className={classes.deleteButton}
+              onClick={() => deleteRule(ruleIdx)}
+            >
+              <CancelIcon fontSize="large" />
+            </IconButton>
+          </If>
+          {rule.map((stage, i) => (
+            <div key={`${i}-${stage.stageName}`} className={classes.stageRow}>
+              <div className={classes.tableCell}>{stage.stageName}</div>
               <div className={classes.tableCell}>
-                <span className={classes.fieldEquality}>=</span>
+                <OutlinedSelect
+                  options={inputSchema[stage.stageName]}
+                  disabled={disabled}
+                  value={stage.fieldName}
+                  onChange={(val) => fieldChange(i, val)}
+                />
               </div>
-            </If>
-          </div>
-        ))}
-      </Paper>
-      <div className={classes.actionButtons}>
-        <If condition={!last}>
-          <span className={classes.andBlock}>AND</span>
-        </If>
-        <If condition={last}>
-          <IconButton disabled={disabled} size="small" onClick={addRule}>
-            <AddIcon fontSize="large" />
-          </IconButton>
-        </If>
+              <If condition={i !== rule.length - 1}>
+                <div className={classes.tableCell}>
+                  <span className={classes.fieldEquality}>=</span>
+                </div>
+              </If>
+            </div>
+          ))}
+        </Paper>
+        <div className={classes.actionButtons}>
+          <If condition={!last}>
+            <span className={classes.andBlock}>AND</span>
+          </If>
+          <If condition={last}>
+            <IconButton disabled={disabled} size="small" onClick={addRule}>
+              <AddIcon fontSize="large" />
+            </IconButton>
+          </If>
+        </div>
       </div>
+      <If condition={error}>
```

Comment:
I'm seeing a TS error here, about the types of condition needing to be boolean (and it is currently being assigned to string?)

Reply:
fixed!